packages/app-headless-cms/src/admin/components/ContentEntryForm/Header/Header.tsx

```diff
@@ -21,19 +26,20 @@ const headerActionsRight = css({
     justifyContent: "flex-start"
 });
 
-const Header: React.FC = props => {
+export const Header: React.FC = () => {
+    const { actions } = useContentEntryEditorConfig();
+
     return (
         <React.Fragment>
             <Grid className={toolbarGrid}>
                 <Cell span={4} className={classNames(headerActions, headerActionsRight)}>
-                    {renderPlugins("cms-content-details-header-left", props)}
+                    <RevisionSelector />
                 </Cell>
                 <Cell span={8} className={classNames(headerActions, headerActionsLeft)}>
-                    {renderPlugins("cms-content-details-header-right", props)}
+                    <Buttons actions={actions.filter(action => action.$type === "button-action")} />
```

Comment:
Move grouping by `$type` into the `useContentEntryEditorConfig`, and expose `buttonActions`, `menuItemActions`.